torch/csrc/jit/api/method.h

```diff
@@ -31,6 +31,11 @@ struct TORCH_API Method {
       std::vector<c10::IValue> stack,
       const Kwargs& kwargs = Kwargs());
 
+  c10::intrusive_ptr<c10::ivalue::Future> run_async(
```

Comment:
add a comment talking about threading model. What runs on the caller thread, when the return happens, etc.